service/controller/clusterapi/v18/key/cluster.go

```diff
@@ -57,6 +57,12 @@ func ClusterName(cluster v1alpha1.Cluster) string {
 	return clusterProviderSpec(cluster).ClusterName
 }
 
+// EncryptionKeySecretName generates name for a Kubernetes secret based on
+// information in given v1alpha1.ClusterGuestConfig.
+func EncryptionKeySecretName(cr v1alpha1.Cluster) string {
```

Comment:
Do we need to have that in the key package? It looks like it's used only by the encryptionkey resource.

Reply:
I like that. Moved to resource as private key function.